go/database/mpt/archive_trie_test.go

```diff
@@ -1690,6 +1684,41 @@ func TestArchiveTrie_FailingLiveStateUpdate_InvalidatesArchive(t *testing.T) {
 	}
 }
 
+func TestArchiveTrie_VisitTrie(t *testing.T) {
+	for _, config := range allMptConfigs {
+		ctrl := gomock.NewController(t)
+		nodeVisitor := NewMockNodeVisitor(ctrl)
+		nodeVisitor.EXPECT().Visit(gomock.Any(), gomock.Any()).AnyTimes()
```

Comment:
I thought `AnyTime()` means atleast once. Fixed.

Anyhow I believe testing whether correct data is passed would mean that the `Forest` is tested instead of the `ArchiveTrie`. I think only call to the `Visitor` should be tested, what data is being passed should be concern of `forest_test.go`

https://github.com/Fantom-foundation/Carmen/blob/deaa6b416d8c69c1ba9ef99a66f35165399f0a12/go/database/mpt/forest.go#L454

Reply:
no, you test as a black-box. You insert accounts and keys, and you test that the visitor returns you nodes representing these accounts and keys. You do not assume anything about forest and layers underneath.